app/services/Payment.js

```diff
@@ -16,19 +17,47 @@ class Payment extends Service {
         return this.fetchJson(url, fetchOptions);
     }
 
+    getCasePayments(data) {
+        this.log('Getting all payments from case');
+        const url = `${this.endpoint}?service_name=Probate&ccd_case_number=${data.caseId}`;
+        const headers = {
+            'Content-Type': 'application/json',
+            'Authorization': data.authToken,
+            'ServiceAuthorization': data.serviceAuthToken
+        };
+        const fetchOptions = this.fetchOptions(data, 'GET', headers);
+        return this.fetchJson(url, fetchOptions);
+    }
+
     post(data, hostname) {
         this.log('Post payment');
         const url = this.endpoint;
         const headers = {
             'Content-Type': 'application/json',
             'Authorization': data.authToken,
             'ServiceAuthorization': data.serviceAuthToken,
-            'return-url': this.formatUrl.format(hostname, '/payment-status')
+            'return-url': this.formatUrl.format(hostname, config.services.payment.paths.returnUrlPath)
         };
         const body = paymentData.createPaymentData(data);
         const fetchOptions = this.fetchOptions(body, 'POST', headers);
         return [this.fetchJson(url, fetchOptions), body.reference];
     }
+
+    identifySuccessfulOrInitiatedPayment(casePayments) {
```

Comment:
As this function is really checking if a payment is made it should return true or false.  A function should always return something.  - This will also remove the check for 'undefined' in the calling routine.
Also, we use a library call lodash which has array processing.  To simplify the array processing you could use the following:
_.indexOf(['Initiated',Iinitiated', 'Initiated', 'Success'], 'Success');

Reply:
The _identifySuccessfulOrInitiatedPayment_ function is bringing back the paymentDto where status is Success or Initiated. In actual fact it only needs the reference so possibily it could be simplified but I'm not sure Lodash indexOf can support key expressions in JSON? So I'd need to create some temporary arrays?

I've created a let variable for an undefinedPayment and I'm returning this if no payment is found, makes it a little easier to follow instead of not returning any value I guess.

Reply:
I guess you could have just had a `return false` at the end of the function, no need to have that let variable.
Then in the check you could just have `if (paymentResponse) {`.